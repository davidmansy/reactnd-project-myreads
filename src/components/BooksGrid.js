import React from 'react';
import BookCard from './BookCard';

function BooksGrid(props) {
  const { books } = props;
  return (
    <ol className="books-grid">
      {books.map(book => {
        return (
          <li key={book.id}>
            <BookCard book={book} />
          </li>
        );
      })}
    </ol>
  );
}

export default BooksGrid;
