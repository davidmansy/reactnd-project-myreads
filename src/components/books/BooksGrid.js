import React, { Component } from 'react';
import BookCard from './BookCard';

class BooksGrid extends Component {
  onDrop = e => {
    const { handleShelfChange, shelfId } = this.props;
    const id = e.dataTransfer.getData('id');
    handleShelfChange({ id: id }, shelfId);
  };

  render() {
    const { books, handleShelfChange } = this.props;

    return (
      <ol
        className="books-grid"
        onDragOver={e => e.preventDefault()}
        onDrop={this.onDrop}
      >
        {books.map(book => {
          return (
            <li key={book.id}>
              <BookCard book={book} handleShelfChange={handleShelfChange} />
            </li>
          );
        })}
      </ol>
    );
  }
}

export default BooksGrid;
