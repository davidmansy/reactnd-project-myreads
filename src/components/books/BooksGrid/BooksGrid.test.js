import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import BooksGrid from './BooksGrid';
import { MemoryRouter } from 'react-router-dom';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

describe('BooksGrid', () => {
  const props = {
    books: [],
    handleShelfChange: () => {},
    draggable: 'false',
    shelfId: 'read'
  };

  it('renders', () => {
    const container = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <BooksGrid {...props} />
      </MemoryRouter>,
      container
    );
  });

  it('snapshots', () => {
    const testRenderer = TestRenderer.create(
      <MemoryRouter>
        <BooksGrid {...props} />
      </MemoryRouter>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
