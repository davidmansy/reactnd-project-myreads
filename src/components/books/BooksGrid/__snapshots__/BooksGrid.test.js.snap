// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksGrid snapshots 1`] = `
<ol
  className="books-grid"
  onDragOver={[Function]}
  onDrop={[Function]}
/>
`;
